src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -182,6 +237,25 @@ public static void disableCache() {
     return resolveRawArguments(resolveGenericType(type, subType), subType);
   }
 
+  /**
+   * The same as {@link #substituteGenerics(Type, Class)}, but first resolves the generic type of {@code type}.
+   * This is a convenience method.
+   */
+  public static <T, S extends T> Type substituteGenerics(Class<T> type, Class<S> subType) {
```

Comment:
A name that came to mind for me is "reifyGenerics", which seems appropriate here :)